### DISCUSSION TITLE
heavy_check_mark is not rendered correctly inside tables

### DISCUSSION THREAD

**Discussion by Bergmann89** [community/community#59743] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

Hey guys,

I have faced the issue that the `heavy_check_mark` is not rendered correctly inside tables. Sometimes it is rendered as UTF-8 character (which displays a grey icon) and sometimes as `g-emoji` html tag (which displays a green icon).

You can finde the code that is rendered incorrectly in [this repository](https://github.com/peeriot/murf/commit/e6746e649122664e7b55459bbc824f1bc8350709).

And it looks like if we face the same problem if I copy the table to this post:

| Feature | murf | mockall | mockers | mock_derive | galvanic_mock | pseudo | faux | unimock | mry |
| --- | --- | --- | --- | --- | --- | --- | --- | --- | --- |
| maintained | :heavy_check_mark: | :heavy_check_mark: | :heavy_plus_sign: | :x: | :x: | :x: | :heavy_plus_sign: | :heavy_check_mark: | :heavy_plus_sign: |
| documentation | :x: (planed) | :heavy_check_mark: | :heavy_check_mark: | :heavy_plus_sign: | :heavy_plus_sign: | :heavy_plus_sign: | :heavy_check_mark: | :heavy_check_mark: | :heavy_plus_sign: |
| proc macro | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :x: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| dev-dependency only | :heavy_check_mark: | :heavy_check_mark: | :heavy_plus_sign: | :heavy_plus_sign: | :heavy_plus_sign: | :heavy_check_mark: | :heavy_plus_sign: | :heavy_plus_sign: | :heavy_plus_sign: |
| matcher / action interface | :heavy_check_mark: | :heavy_check_mark: | :x: | :x: | :heavy_plus_sign: | :heavy_plus_sign: | :heavy_check_mark: | :heavy_check_mark: | :heavy_plus_sign: |
| split into handle and mock object | :heavy_check_mark: | :x: | :heavy_check_mark: | :x: | :x: | :x: | :x: | :x: | :x: |
| support for local references | :heavy_check_mark: | :heavy_check_mark: | :x: | :black_circle: | :black_circle: | :x: | :heavy_plus_sign: | :x: | :black_circle: |
| support for `Self` type | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :black_circle: | :black_circle: | :heavy_check_mark: | :black_circle: | :heavy_check_mark: | :black_circle: |
| support for generic traits | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_plus_sign: | :x: | :heavy_plus_sign: |:heavy_plus_sign: |
| support for associated types | :heavy_check_mark: |:heavy_check_mark: | :heavy_plus_sign: | :black_circle: | :heavy_plus_sign: | :heavy_plus_sign: | :x: | :heavy_plus_sign: | :heavy_plus_sign: |
| support for associated functions | :x: (planed) | :heavy_check_mark: | :x: | :black_circle: | :x: | :heavy_plus_sign: | :x: | :x: | :heavy_check_mark: |
| support for default actions | :heavy_check_mark: | :x: | :x: | :heavy_check_mark: | :x: | :heavy_plus_sign: | :x: | :x: | :heavy_check_mark: |
| define expectations in a sequence | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :black_circle: | :x: | :x: | :x: | :heavy_plus_sign: | :x: |
| support for checkpoints | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :black_circle: | :x: | :heavy_plus_sign: | :x: | :x: | :x: |
| define call count for expectations | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :x: | :heavy_check_mark: | :heavy_plus_sign: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |

Did I make any mistake? Or is there any way to quick fix this within the repository?

Thanks,
Bergmann89.

**Comment by jgneff** [comment]
Yes, I'm seeing the same problem with the README file for the repository [jgneff/openjdk](https://github.com/jgneff/openjdk). All of the check marks in the table below are the same character, but the symbol is rendered differently:

### In dark mode

![Screenshot 2023-07-01 at 16-49-09 jgneff_openjdk Current JDK release and early-access builds](https://github.com/community/community/assets/1413266/e883e6d0-d4d8-45ec-add0-a518a8582574)

### In standard mode

![Screenshot 2023-07-01 at 16-49-34 jgneff_openjdk Current JDK release and early-access builds](https://github.com/community/community/assets/1413266/2174856b-0567-44eb-b56c-d1a8ead2a120)

I'm unable to get GitHub to generate all of the check mark symbols the same way, even using the Unicode variant selectors `U+FE0E` and `U+FE0F`.

**Comment by jgneff** [comment]
I updated my README files to replace the ✔ HEAVY CHECK MARK symbol (U+2714) with the normal ✓ CHECK MARK symbol (U+2713) to avoid the bug. See [my commit](https://github.com/jgneff/openjdk/commit/a75a8b2a001a7d055fc3fcafc16fd6d090acdc14) to the repository [jgneff/openjdk](https://github.com/jgneff/openjdk) for details. With this change, GitHub no longer tries to replace the characters with an equivalent emoji image.

**Comment by Bergmann89** [comment]
Hey, 

for some reason it's working fine now. I did not change anything. Looks like someone fixed this issue without noticing this discussion :)

Thanks

**Comment by mootari** [comment]
I'm seeing the same problem, in this case check marks inside summary elements (screenshot taken in macOS Chrome):

<img width="330" alt="image" src="https://github.com/community/community/assets/322014/ca17c2f5-4fe8-4583-a754-dd2e383f8a4f">

**Comment by Bergmann89** [comment]
Ok, strange. It's happening again. 3 hours ago everything seems to work perfectly, but right now (using the same browser) it's broken again :/

**Comment by omarkohl** [comment]
I'm seeing the same thing within lists. Both the "red X" emoji :x: and the "green heavy checkmark" :heavy_check_mark: are sometimes rendered as Unicode characters and sometimes as HTML. When editing the markdown source the result changes and seems to be completely random. In fact while writing this comment I noticed that the same thing happens within normal markdown text.
:heavy_check_mark:
:heavy_check_mark:
:heavy_check_mark:
:heavy_check_mark:

![image](https://github.com/community/community/assets/1420656/41448fb9-e45a-4204-a5f7-f23eaf717814)

In the screenshot above there is, in this order:
* the Unicode check mark `✔️`
* Generated HTML `<img class="emoji" alt="x" src="https://github.githubassets.com/images/icons/emoji/unicode/274c.png" width="20" height="20">`
* Generated HTML `<img class="emoji" alt="heavy_check_mark" src="https://github.githubassets.com/images/icons/emoji/unicode/2714.png" width="20" height="20">`
* Generated HTML `<img class="emoji" alt="x" src="https://github.githubassets.com/images/icons/emoji/unicode/274c.png" width="20" height="20">`
* Unicode `❌`

**Reply by omarkohl** [reply]
Source of the screenshot: https://github.com/cleodora-forecasting/cleodora/issues/112#issuecomment-1656744107

**Comment by cnlohr** [comment]
Now, it is broken for me.  It was working earlier. Appears for me both on heavycheckmark and whitecheckmark. 
![image](https://github.com/community/community/assets/2748168/47e8831f-e8d8-4634-b5ca-f15a067791c3)

**Comment by mehulrajdev** [comment]
Check this, please

(https://github.com/mehulrajdev/Zayn-Malik))

https://github.com/mehulrajdev
